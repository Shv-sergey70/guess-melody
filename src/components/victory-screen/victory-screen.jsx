import React from 'react';
import {connect} from "react-redux";
import PropTypes from 'prop-types';
import {getMistakes, getWastedTime} from "../../reducer/game/selectors";
import PlayAgainLink from "../links/play-again/play-again";

const getFormattedTime = (time) => {
  return {
    minutes: Math.floor(time / 60),
    seconds: Math.floor(time % 60)
  };
};

const VictoryScreen = ({mistakesCount, wastedTime}) => {
  const formattedTime = getFormattedTime(wastedTime);

  return (
    <section className="login">
      <div className="login__logo">
        <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83"/>
      </div>
      <h2 className="login__title">Вы настоящий меломан!</h2>
      <p className="login__total">За {formattedTime.minutes} минуты и {formattedTime.seconds} секунд вы набрали 12 баллов (8 быстрых), совершив {mistakesCount} ошибки</p>

      <PlayAgainLink />
    </section>
  );
};

VictoryScreen.propTypes = {
  mistakesCount: PropTypes.number.isRequired,
  wastedTime: PropTypes.number.isRequired,
};

const mapStateToProps = (state) => ({
  mistakesCount: getMistakes(state),
  wastedTime: getWastedTime(state)
});

export {VictoryScreen};
export default connect(mapStateToProps)(VictoryScreen);
