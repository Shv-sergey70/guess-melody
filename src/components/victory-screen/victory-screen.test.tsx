import * as React from 'react';
import {configure, shallow} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {VictoryScreen} from "./victory-screen";

configure({adapter: new Adapter()});

test(`Victory screen renders correctly`, () => {
  const tree = shallow(
      <VictoryScreen
        mistakesCount={1}
        wastedTime={75}
        totalScores={12}
        fastScores={8}
      />
  );

  expect(tree).toMatchSnapshot();
});
