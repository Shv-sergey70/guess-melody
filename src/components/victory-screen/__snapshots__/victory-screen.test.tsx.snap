// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Victory screen renders correctly 1`] = `ShallowWrapper {}`;
